packages/start-frontend/src/helpers/prompt-clack.ts

```diff
@@ -0,0 +1,184 @@
+            return {
+              value: key,
+              label: CLIOptions[key].name,
+            };
+          }),
+        }) as Promise<string>;
+      },
+
+      // Select API Solution
+      apiSolution: () => {
+        return select({
+          message: "Select an API Solution (Use arrow keys)",
+          options: CLIOptions.react.apiSolution.map((solution) => {
+            return {
+              value: solution.value,
+              label: solution.name,
+            };
+          }),
+        }) as Promise<"graphql" | "restful">;
+      },
+
+      // Select modules for selected JS Library
+      modules: (value) => {
+        return multiselect({
+          message:
+            "Select the modules that you would like to use (Press 'space' to select)",
+          options: CLIOptions[value.results.jsLibrary as string].useModules,
+          required: false,
+          initialValues: [
+            CLIOptions[value.results.jsLibrary as string].useModules[0].value,
+          ],
+        }) as Promise<string[]>;
+      },
+
+      isUseSampleTestCode: () => {
+        return confirm({
+          message: "Add testing codes for catching bugs early?",
+        });
+      },
+    },
+    {
+      // On Cancel callback that wraps the group
+      // So if the user cancels one of the prompts in the group this function will be called
+      onCancel: () => {
+        cancel("Operation cancelled.");
+        process.exit(0);
+      },
+    }
+  );
+
+  let testGroupUtility: UserInputTests | undefined;
+
+  if (groupUtility.isUseSampleTestCode) {
+    // Select testing packages
+    testGroupUtility = await group(
+      {
+        useVitest: () => {
+          return confirm({
+            message: "Add Vitest for Unit Testing?",
+            initialValue: true,
+          });
+        },
+        useStorybook: () => {
+          return confirm({
+            message: "Add Storybook for Visual Testing?",
+            initialValue: true,
+          });
+        },
+        useE2E: () => {
+          return confirm({
+            message: "Add Playwright for End-To-End Testing?",
+            initialValue: true,
+          });
+        },
+        useEslint: () => {
+          return confirm({
+            message: "Add ESLint for Code Linting?",
+            initialValue: true,
+          });
+        },
+        usePrettier: () => {
+          return confirm({
+            message: "Add Prettier for Code Formatting?",
+            initialValue: true,
+          });
+        },
+      },
+      {
+        // On Cancel callback that wraps the group
+        // So if the user cancels one of the prompts in the group this function will be called
+        onCancel: () => {
+          cancel("Operation cancelled.");
+          process.exit(0);
+        },
+      }
+    );
+  }
+
+  // running spinner after all the prompts are done
+  // test would be failed if no spinner is running
+  const s = spinner();
```

Comment:
This spinner is necessary for passing test.
Somehow test is failed if no spinner with start and stop.(Always get timeout error under cli final test)

I tried to find the reason, but it's really hard to find why.

Reply:
But which asynchronous process should set?
There is no function `downloadCodeFromGithub` in this file, since this file is just for output the result of what user select while cli process.

So, basically there is no need to set spinner here, but somehow the test is failed without spinner.

Reply:
@jinmayamashita 

Oh, I see... understand.
Let me update the code following your example.